src/bootstrap.js

```diff
@@ -32,27 +34,52 @@ async function run(stateManager, algoliaClient, mainIndex, bootstrapIndex) {
   await stateManager.save({
     stage: 'bootstrap',
   });
+  const packagesFetcher = new PackagesFetcher(
+    {
+      limit: config.bootstrapConcurrency,
+      max: config.packagesPrefetchMax,
+    },
+    stateManager
+  );
 
-  const { seq, nbDocs: totalDocs } = await npm.getInfo();
+  const { seq } = await npm.getInfo();
   if (!state.bootstrapLastId) {
     // Start from 0
-    log.info('⛷   Bootstrap: starting from the first doc');
     // first time this launches, we need to remember the last seq our bootstrap can trust
     await stateManager.save({ seq });
     await algolia.putDefaultSettings(bootstrapIndex, config);
   } else {
-    log.info('⛷   Bootstrap: starting at doc %s', state.bootstrapLastId);
+    packagesFetcher.nextKey = state.bootstrapLastId;
   }
 
+  // Firsy sync packagesFetcher
+  Promise.all([
+    await packagesFetcher.syncTotalWithNPM(),
+    await packagesFetcher.syncOffset(),
+  ]);
+
   log.info('-----');
-  log.info(`Total packages   ${totalDocs}`);
+  log.info(`Total packages   ${packagesFetcher.total}`);
+  log.info(`Starting offset   ${packagesFetcher.nextOffset}`);
```

Comment:
what's the difference between nextOffset & nextKey? number & name?

Reply:
key = cursor to paginate
offset = position in the list

We can only paginate with the key, and the offset is only here to say "we are at <offset>/10000000"